contracts/hub/router/src/ibc/receive.rs

```diff
@@ -273,55 +243,52 @@ fn execute_request_pool_creation(
     let mut one_token_already_exists = false;
 
     for token in pair_with_denom.get_vec_token_info() {
-        // Check if token is already validated. Its validated if it has an escrow on sender chain
-        let mut validated_token = ESCROW_BALANCES.has(
-            deps.storage,
-            (token.token.clone(), sender.clone().chain_uid),
-        );
-
-        // Check if token is already present on any chain
-        let range = ESCROW_BALANCES.prefix(token.token).keys_raw(
-            deps.storage,
-            None,
-            None,
-            Order::Ascending,
-        );
-
-        let token_exists_on_any_chain = range.take(1).count() > 0;
-        if token_exists_on_any_chain {
-            one_token_already_exists = true;
-        }
+        let mut registered_denoms = TOKEN_DENOMS
+            .may_load(deps.storage, token.token.clone())?
+            .unwrap_or_default();
 
+        // If its a voucher, then we need to check if this token atleast exist on one of the chains
         if token.token_type.is_voucher() {
             ensure!(
-                token_exists_on_any_chain,
+                !registered_denoms.is_empty(),
                 ContractError::new(
                     "Cannot create pool with voucher token that doesn't exist on any chain"
                 )
             );
-            // Voucher token is valid if it exists on any chain
-            validated_token = true;
+        } else {
+            let token_registered_on_sender_chain = registered_denoms.iter().any(|denom| {
+                denom.chain_uid == sender.chain_uid && denom.token_type == token.token_type
+            });
+            // If its not a voucher, then this token must be present on sender chain with sent denom or its completely new token
+            ensure!(
+                registered_denoms.is_empty() || token_registered_on_sender_chain,
+                ContractError::new(
+                    format!(
+                        "Token: {}:: sCannot use already existing denom without register first",
```

Comment:
Error message can improve, aside from the obvious, `without register first` -> `without registering it first`